ops/model.py

```diff
@@ -3596,3 +3621,60 @@ def _ensure_loaded(self):
         self._notice = self._container.get_notice(self.id)
         assert self._notice.type == self.type
         assert self._notice.key == self.key
+
+
+class CloudSpec:
```

Comment:
We've been starting to use the [`@dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html) decorator, which writes the `__init__` and `__repr__` and other things for us (probably with `frozen=True`). We'd just need the `from_dict` method.